library/splunk_api_test.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import requests
+# import requests_unixsocket
+import json
+
+UDS_SOCKET_PATH = "/var/run/splunk/cli.socket"
+
+def supports_uds():
+    return os.path.exists(UDS_SOCKET_PATH)
+
+# def api_call_uds(method, endpoint, username, password, payload=None, headers=None, verify=False, status_code=None, timeout=None):
+#     session = requests_unixsocket.Session()
+#     url = f"http+unix://{UDS_SOCKET_PATH}{endpoint}"
+#     if headers is None:
+#         headers = {}
+#     headers['Content-Type'] = 'application/json'
+#     auth = (username, password)
+
+#     response = session.request(method, url, headers=headers, auth=auth, data=json.dumps(payload), verify=verify, timeout=timeout)
+#     if status_code is not None and response.status_code not in status_code:
+#         raise ValueError(f"API call failed with status code {response.status_code}: {response.text}")
+#     return response
+
+# update to take svc_port variable
+def api_call_port_8089(method, endpoint, username, password, payload=None, headers=None, verify=False, status_code=None, timeout=None):
+    url = f"https://127.0.0.1:8089{endpoint}"
+    if headers is None:
+        headers = {}
+    headers['Content-Type'] = 'application/json'
+    auth = (username, password)
+
+    response = requests.request(method, url, headers=headers, auth=auth, data=json.dumps(payload), verify=verify, timeout=timeout)
+    if status_code is not None and response.status_code not in status_code:
+        raise ValueError(f"API call failed with status code {response.status_code}: {response.text}")
+    return response
+
+def main():
+    module_args = dict(
+        method=dict(type='str', required=True),
+        url=dict(type='str', required=True),
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True),
+        body=dict(type='dict', required=False),
+        headers=dict(type='dict', required=False),
+        verify=dict(type='bool', required=False),
+        status_code=dict(type='list', required=False),
+        timeout=dict(type='int', required=False)
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        module.exit_json(changed=False)
+
+    method = module.params['method']
+    endpoint = module.params['url']
+    username = module.params['username']
+    password = module.params['password']
+    payload = module.params.get('payload', None)
+    headers = module.params.get('headers', None)
+    verify = module.params.get('verify', False)
+    status_code = module.params.get('status_code', None)
+    timeout = module.params.get('timeout', None)
+
+    if supports_uds():
```

Comment:
default: 8089 port, even if UDS is enabled. need a flag to switch between UDS/HTTP

list of actions:

- build
- MUST deploy with UDS and non-UDS env
- Splunk/UF works as expected